### ISSUE TITLE
Decide on scope

### ISSUE THREAD

**Issue by JJ** [geneura-papers/2020-evostar-distributed-pool#1]
@salvacorts BSc thesis was very extensive; you can check it out here https://github.com/salvacorts/TFG-Thesis We will probably need to stage publication, as long as it's done in the proper context. In this case, we need to focus on the evolutionary part of it. We have two options here
* Since it's a Go implementation of @pacastillo PhD thesis, we could just focus on the evolutionary part of it: G-Prop redux
* Since it's volunteer implementation of it, we could focus on that aspect: farming/pool based models using WASM. 

What do you think?

**Comment by amorag** [comment]
In my view maybe the best option could be separating this in two papers, but both of them need more work, i.e. experiments, in order to be concluded.

However, if the novelty is using WASM to create the distributed model, a paper focused on the improvement of G-Prop should be written as independent of this approach.

Moreover, after having a look at the MT document I think:
- From the results, it seems that just one execution has been conducted in order to compare a sequential approach with the distributed model (and to conclude the latter is better), so at least 10 runs should be performed.
- The state of the art must be improved (maybe redone) considering similar approaches to this proposal in the literature.
- The description of the proposal must be clearly placed in a separate section and well structured. Now it is quite original but also quite hard to follow what it has been designed and implemented. Moreover, several explanations have been done using source code.

**Comment by JJ** [comment]
El jue., 26 sept. 2019 a las 1:30, Antonio Mora (<notifications@github.com>)
escribió:

> In my view maybe the best option could be separating this in two papers,
> but both of them need more work, i.e. experiments, in order to be concluded.
>
> However, if the novelty is using WASM to create the distributed model, a
> paper focused on the improvement of G-Prop should be written as independent
> of this approach.
>
> Moreover, after having a look at the MT document I think:
>
>    - From the results, it seems that just one execution has been
>    conducted in order to compare a sequential approach with the distributed
>    model (and to conclude the latter is better), so at least 10 runs should be
>    performed.
>    - The state of the art must be improved (maybe redone) considering
>    similar approaches to this proposal in the literature.
>
>
Great. I'll create an issue and assign it to you.

**Comment by salvacorts** [comment]
I agree with @amorag, two papers can be written from this work.

As these two papers will have two different scopes, state of the art of both of them should be written regarding what the paper actually talks about.

> From the results, it seems that just one execution has been conducted in order to compare a sequential approach with the distributed model (and to conclude the latter is better), so at least 10 runs should be performed.

Agree.

We also should consider which paper shall be completed first. I think we should rather go for the distributed island/pool model using gRPC and the library itself, and then go for the WASM implementation.

**Comment by JJ** [comment]
It's probably better if we start with the re-implementation of G-Prop. We
need to prove that it works, that it's faster and maybe slightly better.
It's probably very easy to check it on the same sets.

We are dealing in this case with an evolutionary algorithms with many
parameters, and one of the things we will need to do here is to prove that
we're using the best we can.

If, after this, we see that we don't have enough to fill a paper (I don't
think so), we can extend it to the WASM implementation. But for the time
being, I think it's more than enough. This is just a first, initial paper
and the season of writing (and submitting to conferences) has just started.

**Comment by JJ** [comment]
So we agree on the re-implementation of G-Prop for starters? In Go and Wasm? Without distributed things?

**Comment by salvacorts** [comment]
Yes, I think that is a good scope to start given the limited bandwidth we have.

In case time is left, or for future papers, we can continue with the distributed implementation.

**Comment by JJ** [comment]
@salvacorts Check out what's written already and give it a spin to get into focus. The most important thing is to add current results, so that we can check if we need additional experiments.
